compiler/rustc_codegen_gcc/src/back/write.rs

```diff
@@ -78,14 +78,20 @@ pub(crate) unsafe fn codegen(
 
         if config.emit_ir {
             let out = cgcx.output_filenames.temp_path(OutputType::LlvmAssembly, module_name);
-            std::fs::write(out, "").expect("write file");
+            std::fs::write(&out, "").expect("write file");
+            if config.json_artifact_notifications {
+                dcx.emit_artifact_notification(&out, "llvm_ir");
```

Comment:
Are there any tools which depend on artifact notifications of a given type only occurring once?

Reply:
I'm not aware of such tools. I did check `cargo-metadata` before making this change and it seem to simply copy list of files as is.

Reply:
Correction, `cargo` itself consumes this list and ignores it: https://github.com/rust-lang/cargo/issues/13672, `cargo-metadata` simply passes what it gets from cargo.

So to be affected you need to use `rustc` directly, ask it to emit artifact notifications, ask it to emit llvm/asm/etc and fail if previously unknown artifact type occurs, optionally more than once.